prism/src/main/java/org/apache/falcon/resource/AbstractEntityManager.java

```diff
@@ -282,9 +289,56 @@ public synchronized APIResult update(HttpServletRequest request, String type, St
             throw FalconWebException.newException(e, Response.Status.BAD_REQUEST);
         } finally {
             ConfigurationStore.get().cleanupUpdateInit();
+            releaseUpdateEntityLocks(entityName, tokenList);
         }
     }
 
+    private List<MemoryLocks.LockToken> obtainUpdateEntityLocks(Entity entity)
```

Comment:
Assuming a scenario where e1 lock is obtained by t1 and e2 lock is obtained by t2, the further obtainlocks() will fail and no update will happen. Which is what we want. Adding a cleanup() method in case of exceptions should work fine.